app/javascript/actionsPanel/actionsPanel.js

```diff
@@ -0,0 +1,321 @@
+  }
+}
+
+function correctReactedClasses(category) {
+  const upVote = document.querySelector("[data-category='thumbsup']");
+  const downVote = document.querySelector("[data-category='thumbsdown']");
+  const vomitVote = document.querySelector("[data-category='vomit']");
+
+  if (category === 'thumbsup') {
+    downVote.classList.remove('reacted');
+    vomitVote.classList.remove('reacted');
+  } else {
+    upVote.classList.remove('reacted');
+  }
+}
+
+function addReactionButtonListeners() {
+  const butts = Array.from(
+    document.querySelectorAll('.reaction-button, .reaction-vomit-button'),
+  );
+  /* eslint-disable camelcase */
+  butts.forEach((butt) => {
+    butt.addEventListener('click', (event) => {
+      event.preventDefault();
+      const {
+        reactableType: reactable_type,
+        category,
+        reactableId: reactable_id,
+      } = butt.dataset;
+
+      correctReactedClasses(category);
+      butt.classList.toggle('reacted');
+
+      request('/reactions', {
+        method: 'POST',
+        body: { reactable_type, category, reactable_id },
+      });
+    });
+  });
+  /* eslint-enable camelcase */
+}
+
+function clearExpLevels() {
+  Array.from(
+    document.getElementsByClassName('level-rating-button selected'),
+  ).forEach((el) => {
+    el.classList.remove('selected');
+  });
+}
+
+function updateExperienceLevel(currentUserId, articleId, rating, group) {
+  request('/rating_votes', {
+    method: 'POST',
+    body: JSON.stringify({
+      user_id: currentUserId,
+      article_id: articleId,
+      rating,
+      group,
+    }),
+  }).then((response) =>
+    response
+      .json()
+      .then((json) => {
+        if (json.result === 'Success') {
+          clearExpLevels();
+          document
+            .getElementById(`js__rating__vote__${rating}`)
+            .classList.add('selected');
+        } else {
+          // eslint-disable-next-line no-alert
+          alert(json.error);
+        }
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-alert
+        alert(error);
+      }),
+  );
+}
+
+function toggleSubmitContainer() {
+  document
+    .getElementById('adjustment-reason-container')
+    .classList.toggle('hidden');
+}
+
+function clearAdjustmentReason() {
+  document.getElementById('tag-adjustment-reason').value = '';
+}
+
+function renderTagOnArticle(tagName, colors) {
+  const articleTagsContainer = top.document.getElementsByClassName(
+    'tags',
+  )[0];
+
+  const newTag = document.createElement('a');
+  newTag.innerText = `#${tagName}`;
+  newTag.setAttribute('class', 'tag');
+  newTag.setAttribute('href', `/t/${tagName}`);
+  newTag.style = `background-color: ${colors.bg}; color: ${colors.text};`;
+
+  articleTagsContainer.appendChild(newTag);
+}
+
+function adjustTag(el) {
+  const reasonForAdjustment = document.getElementById('tag-adjustment-reason')
+    .value;
+  const body = {
+    tag_adjustment: {
+      // TODO: change to tag ID
```

Comment:
Is this a TODO for this PR or future PR?

Reply:
Yep, TODO for a future PR